broadlink/climate.py

```diff
@@ -238,3 +238,324 @@ def set_schedule(self, weekday: List[dict], weekend: List[dict]) -> None:
+
+        if (payload_debug):
+            data['received_payload'] = payload
+
+        return data
+
+    def set_advanced(self,
+        state: bool,
+        mode: str,
+        target_temp: float,
+        speed: str,
+        swing_v: str,
+        swing_h: str,
+        sleep: bool,
+        display: bool,
+        health: bool,
+        cmnd_0d_rmask: int,
+        cmnd_0e_rmask: int,
+        cmnd_18: int,
+        checksum_lbit: int
+    ) -> bytes:
+        """Set paramaters of unit and return response. All parameters need to be specified.
+        
+        Args:
+            state (bool): power
+            target_temp (float): temperature set point 16<n<32
+            mode (str): cooling, heating, fan, dry, auto
+            speed (str): mute, low, mid, high, turbo (available only for cooling) 
+            swing_h (str): ON, OFF
+            swing_v (str): ON, OFF, 1, 2, 3, 4, 5 (fixed positions)
+            sleep (bool) 
+            display (bool)
+            health (bool)
+            cmnd_0d_rmask (int): unknown
+            cmnd_0e_rmask (int): unknown
+            cmnd_18 (int): unknown
+            checksum_lbit (int): subtracted from the left byte of the checksum
+        """
+        
+        PREFIX = [0x19, 0x00, 0xbb, 0x00, 0x06, 0x80, 0x00, 0x00, 0x0f, 0x00, 0x01, 0x01] # 12B
+        MIDDLE = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] # 13B + 2B checksum
+        SUFFIX = [0, 0, 0, 0, 0] # 5B
+        payload = PREFIX + MIDDLE + SUFFIX
+
+        assert ((target_temp >= 16) and (target_temp <= 32) and ((target_temp * 2) % 1 == 0))
+
+        if (swing_v == 'OFF'):
+            swing_L = 0b111
+        elif (swing_v == 'ON'):
+            swing_L = 0b000
+        else:
+            raise ValueError('unrecognized swing vertical value {}'.format(swing_v))
+
+        if (swing_h == 'OFF'):
+            swing_R = 0b111
+        elif (swing_h == 'ON'):
+            swing_R = 0b000
+        elif (swing_h >= 0 and swing_v <= 5):
+            swing_R = str(swing_v)
+        else:
+            raise ValueError('unrecognized swing horizontal value {}'.format(swing_h))
+
+        if (speed == 'low'):
+            speed_L, speed_R = 0x60, 0x00
+        elif (speed == 'mid'):
+            speed_L, speed_R = 0x40, 0x00
+        elif (speed  == 'high'):
+            speed_L, speed_R = 0x20, 0x00
+        elif (speed == 'mute'):
+            speed_L, speed_R = 0x40, 0x80
+            assert (mode == 'F')
+        elif (speed == 'turbo'):
+            speed_R = 0x40
+            speed_L = 0x20 # doesn't matter
+        elif (speed == 'auto'):
+            speed_L, speed_R = 0xa0, 0x00
+        else:
+            raise ValueError('unrecognized speed value: {}'.format(speed))
+
+        if (mode == 'auto'):
+            mode_1 = 0x00
+        elif (mode == 'cooling'):
+            mode_1 = 0x20
+        elif (mode == 'drying'):
+            mode_1 = 0x40
+            cmnd_0e_rmask = 0x16
```

Comment:
This mask is probably eliminating the target temperature parameter here.
```python3
>>> bin(0x16)
'0b10110'
```

Reply:
Works both with `cmnd_0e_rmask = 0x16` and `cmnd_0e_rmask = 0xd`, so I removed it. I think I added this because of errors that were actually related to swing_R.